gmso/core/bead.py

```diff
@@ -0,0 +1,137 @@
+import warnings
+from typing import Optional, Union
+
+import unyt as u
+
+from pydantic import Field, validator
+
+from gmso.core.element import Element
+from gmso.abc.abstract_site import Site
+from gmso.core.atom_type import AtomType
+from gmso.utils.misc import ensure_valid_dimensions
+from gmso.utils._constants import UNIT_WARNING_STRING
+
+
+class Bead(Site):
+    __base_doc__ = """A bead represents a single coarse-grained association in a topology.
+    
+    Beads are the representation of an object within `gmso` that describes any general 
+    non-atomistic particle in gmso. Beads also contain information that are unique to 
+    elements vs other types of interaction sites in molecular simulations. 
+
+    Notes
+    -----
+    Beads have all the attributes inherited from the base Site class,
+    The order of precedence when attaining properties `charge` and `mass` is:
+        
+        1. atom.charge > atom.atom_type.charge 
+        2. atom.mass > atom.atom_type.mass
+        
+    Examples
+    --------
+        >>> from gmso.core.bead import Bead
+        >>> bead1 = Bead(name='SP1')
+
+    See Also
+    --------
+    gmso.abc.AbstractSite
+        An Abstract Base class for implementing site objects in GMSO. The class Bead bases from
+        the gmso.abc.abstract site class
+    """
+    charge_: Optional[Union[u.unyt_quantity, float]] = Field(
+        None,
+        description='Charge of the bead',
+    )
+
+    mass_: Optional[Union[u.unyt_quantity, float]] = Field(
+        None,
+        description='Mass of the bead'
+    )
+
+    bead_type_: Optional[AtomType] = Field(
+        None,
+        description='AtomType associated with the atom'
```

Comment:
```suggestion
        description='AtomType associated with the bead'
```
should this be bead?

Reply:
Yes, good catch @jennyfothergill. I'm going to go through and try to write some more implementation for the bead class. Based on the properties you mentioned above, looks like it will act relatively similarly to the current `site` class.